Add specific detail to Ruby gem versions (#4548)
alphagov/govuk-developer-docs: source/manual/publishing-a-ruby-gem.html.md

## Ruby version compatibility

Ensure gems are compatible with all [supported minor Ruby
versions][supported-rubies], specifying the minimum minor Ruby version in the gemspec

> file (for example: `spec.required_ruby_version = ">= 3.1"`) - note, this should not
> be more specific than a minor Ruby version unless there is a particular issue with
> a Ruby patch release.

Reviewer: minor edit for clarity / tech writing style: (avoid "note", prefer callout, avoid bracketed clauses other than for acronym expansions etc.)

```suggestion
file. For example: `spec.required_ruby_version = ">= 3.1"`.

> Do not specify a patch (or "tiny") Ruby version unless there is a particular issue with a Ruby patch release.
```
